mpp/shmem_c_func.m4

```diff
@@ -112,48 +112,77 @@ SHMEM_C_GET_N_NBI(mem,1);
 
 /* AMO: Atomic Swap Routines */
 define(`SHMEM_C_SWAP',
-`SHMEM_FUNCTION_ATTRIBUTES $2 SHPRE()shmem_$1_swap($2 *target, $2 value, int pe)')dnl
+`SHMEM_FUNCTION_ATTRIBUTES $2 SHPRE()shmem_$1_swap($2 *target, $2 value, int pe) __attribute__ ((deprecated))')dnl
 SHMEM_DECLARE_FOR_EXTENDED_AMO(`SHMEM_C_SWAP')
 
+define(`SHMEM_C_ATOMIC_SWAP',
+`SHMEM_FUNCTION_ATTRIBUTES $2 SHPRE()shmem_$1_atomic_swap($2 *target, $2 value, int pe)')dnl
+SHMEM_DECLARE_FOR_EXTENDED_AMO(`SHMEM_C_ATOMIC_SWAP')
+
 /* Special case, only enabled when C++ and C11 bindings are disabled */
 #if !defined(__cplusplus) && \
     !(defined(__STDC_VERSION__) && __STDC_VERSION__ >= 201112L)
 long SHPRE()shmem_swap(long *target, long value, int pe);
+long SHPRE()shmem_atomic_swap(long *target, long value, int pe);
```

Comment:
We didn't add a ```shmem_atomic_swap``` routine in the C language bindings (only the generic version in C11).  The above new special case should be deleted.